### PR TITLE
fix: pre_hook (sync) make request mutable

### PR DESCRIPTION
**Pull request by drahnr**
Obsoletes #823

### CODE REVIEW

**Review by ahl — COMMENTED** [review]
I'm sort of surprised that we don't have a test that validates this. What will existing users of `pre_hook` experience other than a type error if they've specified the type in their pre_hook closure or fn?

**Comment by drahnr** [comment]
> I'm sort of surprised that we don't have a test that validates this. What will existing users of `pre_hook` experience other than a type error if they've specified the type in their pre_hook closure or fn?

Can you rephrase, I am not sure I understand what your concern is. Do you want an interface check for the provided `TokenStream` or are you concern re change in behviour?

**Comment by ahl** [comment]
If there is an existing user of the `pre_hook` interface, what will those users experience with this change? Will everything just work or will compilation fail with an error?

**Comment by drahnr** [comment]
Happy new year!

Mildly belated response; the user won't see a delta, since a we can call any `fn foo(&T)` with `foo(&mut t)`.

**Review by ahl — APPROVED** [review]
thanks for this; sorry for the very long delay
